### ISSUE TITLE
Service worker expiration

### ISSUE THREAD

**Issue by TomPradat** [w3c/ServiceWorker#1480]
I was wondering if there is some spec about the expiration of a service-worker in the browser. If not this could result in the following problem :

Let's say I decide to create a website and cache all the content with a service-worker (except for the service-worker file) so that the website is available offline.

If someone buy the same domain name later and adds a website without a service-worker, the people that previously went on my website would never see the new website (unless they clear the service-worker by hand). What's more the new owner of the domain name would not even know about it

Linked to https://github.com/mozilla/serviceworker-cookbook/issues/327
PS : I talked about this in #1366 but never got an answer

**Comment by TomPradat** [comment]
Ping @jakearchibald @slightlyoff

**Comment by jakearchibald** [comment]
We talked about removing service workers on 404 in https://github.com/w3c/ServiceWorker/issues/204#issuecomment-342658175.

> If someone buy the same domain name later and adds a website without a service-worker

This isn't a problem for service workers only. It's also an issue with cookies and origin storage. The overall solution is https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Clear-Site-Data.

If you want to target service workers specifically, requests for service worker scripts are sent with a `Service-Worker: script` header. Servers could react to that and serve a service worker which uninstalls itself.

We did a stupider version of this on the Chrome Dev Summit website, where any 404 ending in `.js` serves a self-destructing service worker: https://developer.chrome.com/devsummit/does-not-exist.js
